packages/whale-api-client/__tests__/api/governance.test.ts

```diff
@@ -322,6 +322,19 @@ describe('governance - listProposalVotes', () => {
     expect(result.length).toStrictEqual(3)
   })
 
+  it('should listProposalVotes with pagination', async () => {
+    const result = await client.governance.listGovProposalVotes({ id: cfpProposalId, size: 2, masternode: MasternodeType.ALL, cycle: -1 })
+    expect(result.length).toStrictEqual(2)
+    expect(result.hasNext).toStrictEqual(true)
+    const result1 = await client.governance.listGovProposalVotes({ id: cfpProposalId, size: 2, next: result.nextToken, masternode: MasternodeType.ALL, cycle: -1 })
+    expect(result1.length).toStrictEqual(2)
+    expect(result1.hasNext).toStrictEqual(true)
+    const result2 = await client.governance.listGovProposalVotes({ id: cfpProposalId, size: 2, next: result1.nextToken, masternode: MasternodeType.ALL, cycle: -1 })
+    expect(result2.length).toStrictEqual(0)
```

Comment:
Seems odd that `result1.hasNext` is `true`, but calling the API again with `result1.nextToken` gives a list of length 0 🤔 

Might be a natural limitation of how the cursor pagination is currently implemented?
Maybe we want to take a look at how `hasNext` is computed

Reply:
This is indeed the current implementation, where the pagination response doesn’t perform a +1 look ahead.

This is the related issue I created last time https://github.com/JellyfishSDK/jellyfish/issues/1490 and to quote fx: 

> “..it’s expected a tradeoff when it comes with sort keys LSM querying.”

Reply:
Yeah, jellyfish LSM implementation doesn't do a +1 look ahead. Maybe we should, but it's not really a priority here. Since the clients are "taught" or rather forced to handle this.

> The purest lowest form of sorted sets implementation technically doesn't do it either since the last/next is always the last object; you can always create the "next" token yourself if you know how to do it. The "next" is merely a helper/suggestion.